python/lsst/afw/geom/linearTransform.cc

```diff
@@ -57,6 +57,20 @@ PYBIND11_PLUGIN(_linearTransform) {
     clsLinearTransform.def(py::init<typename LinearTransform::Matrix const &>());
 
     /* Operators */
+    clsLinearTransform.def("__mul__", [](LinearTransform const & self,
+                                         LinearTransform const & other) {
+            return self*other;
+        }, py::is_operator());
+    // __getitem__ implemented in linearTransform.py
+    clsLinearTransform.def("_getitem_nochecking", [](LinearTransform const & self,
+                                                     int i) {
+            return self[i];
```

Comment:
I suggest you implement this using `cppIndex` in `pybind11.h` in the `utils` package. It saves a lot of trouble because it handles negative indices in the usual Python way and does full range checking with a nice verbose error message. That's how we do almost all of our Python indexing. It raises `lsst::pex::exceptions::OutOfRangeError` which I realize is a change (so a minor tweak to a test or two may be needed), but as I say...it's what most of our code does. This is all you need (edited as you like), where `2` is the number of elements:

    { auto cind = utils::cppIndex(2, i); return self[i]; }

And similarly for the next overload of this method but you have to pass the appropriate dimension to each call to `cppIndex`...